### ISSUE TITLE
fix: NF4 LoRA weights never updated with gradient_accumulation > 1

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#264]
## Bug

When `accumulation_steps > 1`, the NF4 LoRA optimizer step was gated behind `if !accumulate_only`, which was always `false` for micro-batches in gradient accumulation mode. This meant LoRA adapter weights (A and B matrices) stayed at their initialization values (B=0) and were **never updated** during training.

## Root Cause

In `gpu_backward()`, the NF4 LoRA optimizer code at line 1449:
```rust
if !accumulate_only {
    // LoRA optimizer step
}
```

`accumulate_only` is set to `true` whenever `grad_accum.is_some()`, which is always true when `accumulation_steps > 1`.

## Impact

- Training appeared to work (grad_norm was non-zero from LM head gradients)
- But LoRA adapters were frozen — only LM head, final norm, and embeddings were trained
- Run 7 SSC training (Qwen3-4B NF4 QLoRA) was affected

## Fix

Commit b9873c8: Run LoRA optimizer every micro-batch with `lr / accumulation_steps`, which is mathematically equivalent to accumulating then stepping with full lr.

**Comment by noahgift** [comment]
Fixed in b9873c8. LoRA optimizer now runs every micro-batch with lr/accumulation_steps.
